### ISSUE TITLE
ufw_forwarding variable undefined

### ISSUE THREAD

**Issue by zxyz** [EggieCode/ansible-role-ufw#2]
> 'ufw_forwarding' is undefined\n\nThe error appears to have been in /roles/ufw/tasks/main.yml

`default/main.yml` (excerpt):

```
ufw:
  forwarding: []
```

which becomes (if I understand it correctly) `ufw.forwarding` which is also in the readme example.

tasks/main.yml (excerpt):

```
 ufw_forward:
data: "{{ ufw_forwarding }}"
```
here ufw_forwarding is referenced. 

(BTW: just a simple example for lets say NAT64 on host A port 80 toVM-B port 8080 might be helpfull for the unexperienced reader)

Thanks for your work & help!

**Comment by EggieCode** [comment]
Changed the variables from `ufw.forwarding` to `ufw_forwarding` was more convenient.
Because the ufw_forward.py lib only generated the iptable roles, and ansbile use the output of that for the template of the before roles (/etc/ufw/before.rules & /etc/ufw/before6.rules).

Updated: 57202be

PS.: Thanks issues.

An example from a server:
```yaml
ufw_forwarding:
  - comment: "Home"
    incomming_dev: enp2s0
    outgoing_dev: locbr0
    outgoing_network: 10.21.0.0/24
    masquerading: True
    conntrack_state: RELATED,ESTABLISHED
    reroute: []
    forwards: 
     - destination_ip: 10.21.0.12
       destination_port: 49999
       incomming_port: 49999
       protocol: ['tcp', 'udp']

ufw_forwarding6:
  - incomming_dev: he-ipv6
    outgoing_dev: locbr0
    outgoing_network: 2001:470:xx:xx::/64
    conntrack_state: RELATED,ESTABLISHED
    forwards:
      - comment: SSH
        destination_network: 2001:470:xx:xx::/64
        destination_port: 22
        protocol: tcp
```

**Comment by zxyz** [comment]
thanks, now the provisioning is working. 
Now I only have to deploy it to a real system and not to vagrant :)

**Comment by zxyz** [comment]
Dear Eggie,
I need your help again please. I have to do IPv6 port forwarding/NAT (yes I know, really ugly but it's a tricky situation caused due to the IPv4 NAT and a reverse proxy on a VM fronting other VM web services which now also has to front the web services on IPv6 :( ).

I accomplished the NAT66 port forwarding by trial and error, resulting in the following ip6tables rules.

```
ip6tables -t nat -A PREROUTING -p tcp -i eth0 --dport 443 -j DNAT --to 2001:db8:1:2::3
ip6tables -t nat -A POSTROUTING -p tcp --dport 443 -j MASQUERADE
```

unfortunaltey I'm not able to produce this rules (or other working ones) with the ansible role. In your example above you used `masquerading: True` but the masquerading feature doesn't seem to work on IPv6. One of the confs I tried:

```
ufw_forwarding6:
  - comment: FWD SSL web traffic
    incomming_dev: eth0
    outgoing_network: 2001:db8:1:2::3
    conntrack_state: RELATED,ESTABLISHED
    forwards:
      - comment: WEB SSL
        destination_network: 2001:db8:1:2::3/128
        destination_port: 443
        protocol: tcp
```

any help would be very much apprechiated!

**Comment by EggieCode** [comment]
Yes, got it.
Simple answer is use `destination_ip` instateof `destination_network`.
The lib support both but `destination_network` **only** will work with a routed IPv6/IPv4 subnet.
Add `incomming_port` because it need a source port, like IPv4 source and destination.

Sorry for the bad names, translation from Dutch.
Example: https://mirrors.bieringer.de/Linux+IPv6-HOWTO/nat-netfilter6..html
Sidenote: Got the same problem with other roles with ufw.

**Comment by EggieCode** [comment]
README fixed: f77a7f8

**Comment by zxyz** [comment]
if I ommit `outgoing_network: 2001:db8:1:2::3` and therfore there is no `-s  2001:db8:1:2::3`int eh POSTROUTING MASQUERADE rule it works. thanks a lot!
